Links to UAVCAN v1.0 (#45)
UAVCAN/uavcan.github.io: index.md

layout: home
---

**UAVCAN is an open lightweight protocol designed for reliable communication in aerospace and robotic

> applications over robust vehicular networks such as CAN bus.**

Reviewer: 👍

- Democratic network -- no bus master, no single point of failure.
- Publish/subscribe and request/response (RPC) exchange semantics.
- Efficient exchange of large data structures with automatic decomposition and reassembly.
- Lightweight, deterministic, easy to implement, and easy to validate.
- Suitable for deeply embedded, resource constrained, hard real-time systems.
- Supports dual and triply modular redundant transports.
- Supports high-precision network-wide time synchronization.
- High-quality open source reference implementations are freely available under the MIT license.

This website is dedicated to the **experimental version** of the protocol, otherwise known as **UAVCAN v0**.

> There is work underway to release the first stable long-term protocol specification and a set of

Reviewer: ```suggestion
There is work underway to release the first version of the protocol specification, based on the learnings from v0, and a set of
```

> robust reference implementations by **2019 Q2**, at which point UAVCAN v0 will reach the end of its life cycle.

Reviewer: I'm not on board for v0 "reaching the end of its life-cycle" in Q2 of 2019. I think it has reached the end of active development already and, by the transitive property, so have the reference implementations but the standard will only be "superseded" by the new standard. "end of lifecycle" implies that existing deployments are obsolete which, as the owner of an existing deployment, is somewhat offensive.

After defining that v1 supersedes v0 we can discuss v0 "end of life-cycle" as being at least 2-years out (2021). We should also clarify that the short EOL timeframe is because of the experimental nature of UAVCAN v0 and that v1 will receive longer Long Term Support (to borrow the Ubuntu "LTS" acronym).

Finally, let's not give the false impression that the v0 specification has any _fatal_ defects. While many of the standard datatypes haven't provided the utility expected the core protocol and some of the standard datatypes have proven to be completely functional.

Author: Please review.

The new stable version that will replace it is known as **UAVCAN v1.0**.

> **All new designs should avoid reliance on v0 and use UAVCAN v1.0 instead.**

Reviewer: ```suggestion
**All new designs should use UAVCAN v1.0.**
```

Existing deployments will benefit from our commitment to provide long-term support and maintenance of

> UAVCAN v0-based systems even after its EOL.

Reviewer: ```suggestion
UAVCAN v0-based systems.
```
